client/trino-jdbc/src/test/java/io/trino/jdbc/TestJdbcConnection.java

```diff
@@ -670,4 +684,20 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
             return table.build().cursor();
         }
     }
+
+    private static void assertThatFutureIsBlocked(Future<?> future)
+    {
+        if (!future.isDone()) {
+            return;
+        }
+        try {
+            future.get();
```

Comment:
if the future is "blocked", this should just hang and never return

seems like this method could be replaced with
```
verify(!future.isDone(), "Future is done too early");
```

if your intent is to return future completion status (value or exception) when it's blocked, this would require a comment, but the code is not doing this anyway

Reply:
If the future is blocked we won't get here as we will return early 3 lines above. And if it is finished with an error I would like to know the error.